### PR TITLE
feat: add DocSearch as recommended by vuepress

### PR DESCRIPTION
**Pull request by s-pace**
👋 team,

I am working on DocSearch. [We have an integration for Vuepress](https://docusaurus.io/docs/en/search#enabling-the-search-bar). We thought it is a shame you can not also used this search that you can [find on vuejs for example](https://vuejs.org/).

This PR will add DocSearch to the documentation website. It will allow a user to have a learn-as-you-type experience by displaying results thanks to a dropdown in a live way.

Let me know if you need anything.

### CODE REVIEW

**Comment by s-pace** [comment]
I will need some help since the website is not exactly vuepress.

@samajammin  can you help me please 🙏 ?

**Comment by s-pace** [comment]
Should be good. I let you style it as you see fit

File: docs/.vuepress/theme/index.js [hunk 1]
```diff
@@ -1,11 +1,29 @@
-const path = require('path');
+const path = require('path')
```
Comment (s-pace) [inline docs/.vuepress/theme/index.js:1]:
```suggestion
const path = require('path');
```

File: docs/.vuepress/theme/index.js [hunk 2]
```diff
@@ -1,11 +1,29 @@
...
-};
+    plugins: [
+      '@vuepress/register-components',
+      {
+        componentsDir: [path.resolve(__dirname, 'components')]
+      },
+      '@vuepress/active-header-links'
+    ]
+  }
+}
```
Comment (s-pace) [inline docs/.vuepress/theme/index.js:29]:
```suggestion
};
```

**Comment by samajammin** [comment]
Hey @carlfairclough - do you mind looking into this for a bit when you're able? Given your familiarity with the latest search styles, I believe you could finish the final polish on this fairly quickly.

Background: we want to improve our search functionality (#804) & Algolia solves this.

Unfortunately, this update also broke some of our styling:

1) Mobile search is broken

Image 2020-04-13 at 1 54 53 PM

I believe updating AlgoliaSearchBox.vue to use the mobile styles similar to SearchBox.vue will solve for this.

2) Search result styles need to be updated to our colors:
Image 2020-04-13 at 2 15 19 PM
I believe some minor updates to the styles in AlgoliaSearchBox.vue (i.e. fixing variables & adding dark mode colors) will solve this.

Please let me know if you have any questions! Thanks in advance 😄

**Comment by samajammin** [comment]
Thanks again for your work on this @s-pace - I couldn't commit directly to your fork so I'm closing this PR out in favor of #977

**Comment by samajammin** [comment]
@all-contributors please add @s-pace for code

**Pull request closed without merging**